internal/resources/snapshot_backup.go

```diff
@@ -0,0 +1,620 @@
+package resources
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-framework/types/basetypes"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+
+	"github.com/couchbasecloud/terraform-provider-couchbase-capella/internal/api"
+	"github.com/couchbasecloud/terraform-provider-couchbase-capella/internal/api/snapshot_backup"
+	"github.com/couchbasecloud/terraform-provider-couchbase-capella/internal/errors"
+	providerschema "github.com/couchbasecloud/terraform-provider-couchbase-capella/internal/schema"
+)
+
+const (
+	errorMessageWhileSnapshotBackupCreation = "There is an error during snapshot backup creation. Please check in Capella to see if any hanging resources" +
+		" have been created, unexpected error: "
+	errorMessageWhileSnapshotBackupUpdate = "There is an error during snapshot backup update. Please check in Capella to see if any hanging resources" +
+		" have been created, unexpected error: "
+	errorMessageConfigure = "Expected *ProviderSourceData, got:"
+)
+
+var (
+	_ resource.Resource                = &SnapshotBackup{}
+	_ resource.ResourceWithConfigure   = &SnapshotBackup{}
+	_ resource.ResourceWithImportState = &SnapshotBackup{}
+)
+
+type ID struct {
+	ID string `json:"id"`
+}
+
+type IDList struct {
+	Data []ID `json:"data"`
+}
+
+type SnapshotBackup struct {
+	*providerschema.Data
+}
+
+func NewSnapshotBackup() resource.Resource {
+	return &SnapshotBackup{}
+}
+
+func (s *SnapshotBackup) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_snapshot_backup"
+}
+
+func (s *SnapshotBackup) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = SnapshotBackupSchema()
+}
+
+// ImportState imports a remote backup that is not created by Terraform.
+func (s *SnapshotBackup) ImportState(ctx context.Context, req resource.ImportStateRequest, resp *resource.ImportStateResponse) {
+	// Retrieve import ID and save to id attribute
+	resource.ImportStatePassthroughID(ctx, path.Root("id"), req, resp)
+}
+
+func (s *SnapshotBackup) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var plan providerschema.SnapshotBackup
+	diags := req.Plan.Get(ctx, &plan)
+	resp.Diagnostics.Append(diags...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	var (
+		organizationId = plan.OrganizationId.ValueString()
+		projectId      = plan.ProjectID.ValueString()
+		clusterId      = plan.ClusterID.ValueString()
+	)
+
+	createSnapshotBackupRequest := snapshot_backup.CreateSnapshotBackupRequest{
+		Retention:     int(plan.Retention.ValueInt64()),
```

Comment:
why not make `Retention` an int64 ?  can avoid conversion, which looks uneccessary here